### PR TITLE
feat: Add GoogleAnalytics 4 to frontend-platform

### PR DESCRIPTION
**Pull request by UvgenGen**
**Description:**
alternative implementation for https://github.com/openedx/frontend-platform/pull/471

Added support for GA4 to frontend-platform:
- implemented GoogleAnalyticsLoader class
- added loadExternalScripts function to  inititalize.js
- Added tests for GoogleAnalyticsLoader.
- The Google Analytics ID can be configured in the edx-platform MFE_CONFIG settings:
```
MFE_CONFIG = {
    "GOOGLE_ANALYTICS_4_ID": "G-TEST"
}
```
or, the ID can be set in the frontend-platform .env file as:
`GOOGLE_ANALYTICS_4_ID="G-TEST"`

**Merge checklist:**

- [ ] Consider running your code modifications in the included example app within `frontend-platform`. This can be done by running `npm start` and opening http://localhost:8080.
- [ ] Consider testing your code modifications in another local micro-frontend using local aliases configured via [the `module.config.js` file in `frontend-build`](https://github.com/openedx/frontend-build#local-module-configuration-for-webpack).
- [ ] Verify your commit title/body conforms to the conventional commits format (e.g., `fix`, `feat`) and is appropriate for your code change. Consider whether your code is a breaking change, and modify your commit accordingly.

**Post merge:**

- [ ] After the build finishes for the merged commit, verify the new release has been pushed to [NPM](https://www.npmjs.com/package/@edx/frontend-platform).

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @UvgenGen!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Review by davidjoy — APPROVED** [review]
Gorgeous. 🤩 

One tiny nit about an `async` keyword, but this looks great.  I'm approving so you don't feel like you need a re-review, but I do think we should remove the `async` keyword from `loadExternalScripts`.

**Comment by UvgenGen** [comment]
@davidjoy Thank you) I removed the `async` keyword from `loadExternalScripts` and rebased PR

**Comment by UvgenGen** [comment]
@bmtcril no changes needed for edx-platform. We use two settings:
- `GOOGLE_ANALYTICS_4_ID = "G-test"` - for platform
-  for MFE's:
```
MFE_CONFIG = {
    "GOOGLE_ANALYTICS_4_ID": "G-test"
}
```

**Comment by bmtcril** [comment]
Thanks for all of the work on this, Adolfo is away for a bit so I don't think there's any need to wait on his review. 👍

**Review by arbrandes — APPROVED** [review]
Looks good!

**Comment by openedx-webhooks** [comment]
@UvgenGen 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-platform&entry.867055334=https://github.com/openedx/frontend-platform/pull/472&entry.1484655318=https://github.com/UvgenGen&entry.752974735=2023-04-10+14:09&entry.1917517419=2023-04-18+19:36&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
